once_test.py

```diff
@@ -190,6 +192,41 @@ def counting_fn(*args) -> int:
     return counting_fn, counter
 
 
+class LineCapture:
+    def __init__(self):
+        self.line = None
+
+    def record_next_line(self):
+        """Record the next line in the parent frame"""
+        self.line = inspect.currentframe().f_back.f_lineno + 1
+
+
+class ExceptionContextManager:
+    exception: Exception
+
+
+@contextlib.contextmanager
+def assertRaisesWithLineInStackTrace(test: unittest.TestCase, exception_type, line: LineCapture):
+    try:
+        container = ExceptionContextManager()
+        yield container
+    except exception_type as exception:
+        container.exception = exception
+        traceback_exception = traceback.TracebackException.from_exception(exception)
+        if not len(traceback_exception.stack):
+            test.fail("Exception stack not preserved. Did you use the raw assertRaises by mistake?")
```

Comment:
If there was a raw assertRaises, wouldn't we never get to this failure in the first place?